### PR TITLE
feat:  added auth, dark & light theme, and responsiveness

### PR DESCRIPTION
**Pull request by paarthbajaj**

### CODE REVIEW

File: src/components/Home.js [hunk 1]
```diff
@@ -5,15 +5,34 @@ import { useQuiz } from "../context/QuizContext";
...
-  const { quizDispatch, quizState } = useQuiz();
+  const { quizDispatch, quizState, navigate } = useQuiz();
   return (
     <div>
       <header className="quiz-header txt-bold txt-center pt-1">
-        {/* <div>
-        <i class="fas fa-sun"></i>
-          <i className="fas fa-sign-out"></i>
-          <i class="fas fa-moon"></i>
-        </div> */}
```
Comment (neetukumari4858) [inline src/components/Home.js:16]: Remove comments

**Review by neetukumari4858 — APPROVED** [review]
Code is readable
Colour Theme is looking good.
folder is well Structured
Functionality is working fine
overall looks good only one suggestion is to remove commented code

File: src/components/Quiz.css [hunk 2]
```diff
@@ -130,3 +130,36 @@
...
+  .quiz-header li {
+    margin-top: 0.5rem;
+  }
+  .rules-popup-content {
+    width: 90%;
+  }
+  .question-card {
+    width: 90%;
+  }
+  .result-card {
```
Comment (t007rushi) [inline src/components/Quiz.css:162]: suggestion: can make a single class and append to all for DRY

**Review by t007rushi — APPROVED** [review]
Good naming with nice folder structure 
Good work keep it up!

**Pull request merged**